### PR TITLE
Aql cleanup

### PR DESCRIPTION
**Pull request by stanistan**

### CODE REVIEW

File: lib/Sky/AQL/Exception/Connection.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Sky\AQL\Exception;
+
+class Connection extends \Sky\AQL\Exception
```
Comment (will123195) [inline lib/Sky/AQL/Exception/Connection.php]: ConnectionException

File: lib/Sky/AQL/Exception/Transaction.php [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        return (is_array($this->fields)) ? 'update' : 'increment';
+    }
+
+    /**
+     * Gets the SQL from the insert / update
+     * we can only auto generate the sql in this scenario
+     * @param   ADODB_postgres7 $db
+     * @return  string
+     */
+    private function getSQL(\ADODB_postgres7 $db)
```
Comment (will123195) [inline lib/Sky/AQL/Exception/Transaction.php]: remove database version dependency

File: lib/Sky/AQL/Exception/Transaction.php [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+     * @var int
+     */
+    public $id = null;
+
+
+    /**
+     * @param   string          $table
+     * @param   mixed           $fields
+     * @param   mixed           $id     can be null
+     * @param   ADODB_postgres7 $db
```
Comment (will123195) [inline lib/Sky/AQL/Exception/Transaction.php]: mixed adodb object type

File: lib/class/class.AQLSelectException.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
```
Comment (will123195) [inline lib/class/class.AQLSelectException.php]: delete file

File: lib/class/class.aql.php [hunk 5]
```diff
@@ -1,885 +1,652 @@
...
+     * @param   mixed   $aql
+     * @param   mxied   $clause
+     * @param   mixed   $obj
+     * @param   mixed   $statement
+     * @param   Boolean $force_db
+     */
+    public static function selectDBW($aql, $clause = null, $obj = false, $statement = null, $force_db = false)
+    {
+        $dbw = self::getMasterDB();
+        return aql::select($aql, $clause_array, $object, $aql_statement, $force_db, $dbw);
```
Comment (will123195) [inline lib/class/class.aql.php]: $clause, fix names

File: lib/class/class.aql.php [hunk 6]
```diff
@@ -888,40 +655,694 @@ public function value($param1, $param2, $options = array()) {
...
+
+    /**
+     * @param   array   $arr    generated sql array
+     * @param   array   $settings
+     * @param   db      $db_conn
+     * @return  array
+     * @throws  \Sky\AQL\Exception\Connection   if no db
+     * @throws  \Sky\AQL\Exception\Select       if db select fails
+     */
+    public static function sql_result($arr, $settings, $db_conn = null)
```
Comment (will123195) [inline lib/class/class.aql.php]: private

File: lib/class/class.aql.php [hunk 7]
```diff
@@ -888,40 +655,694 @@ public function value($param1, $param2, $options = array()) {
...
+     * @param   string  $aql
+     * @return  Boolean
+     */
+    public static function is_aql($aql)
+    {
+        return strpos($aql, '{') !== false;
+    }
+
+    /**
+     * @param   array   $arr    generated sql array
```
Comment (will123195) [inline lib/class/class.aql.php:971]: description

**Comment by stanistan** [comment]
This is updated.

**Pull request merged**